orb-info/src/orb_token.rs

```diff
@@ -0,0 +1,165 @@
+use color_eyre::eyre::Result;
+use orb_attest_dbus::AuthTokenManagerProxy;
+use std::sync::Arc;
+use tokio::sync::watch;
+use tokio_util::sync::CancellationToken;
+use zbus::export::futures_util::StreamExt;
+use zbus::Connection;
+
+use crate::OrbInfoError;
+
+#[derive(Debug, Clone)]
+pub struct OrbToken {
+    pub token_receiver: watch::Receiver<String>,
+    pub join_handle: Arc<tokio::task::JoinHandle<()>>,
+    pub cancel_token: CancellationToken,
+}
+
+impl OrbToken {
+    pub async fn read(cancel_token: &CancellationToken) -> Result<Self, OrbInfoError> {
+        let cancel_token = cancel_token.child_token();
+        let (join_handle, token_receiver) =
+            Self::setup_dbus(cancel_token.clone()).await?;
+        Ok(OrbToken {
+            token_receiver,
+            join_handle: Arc::new(join_handle),
+            cancel_token,
+        })
+    }
+
+    pub async fn value(&self) -> Result<String, OrbInfoError> {
+        let reply = self.token_receiver.borrow().to_owned();
+        Ok(reply.to_string())
+    }
+
+    async fn setup_dbus(
+        cancel_token: CancellationToken,
+    ) -> Result<(tokio::task::JoinHandle<()>, watch::Receiver<String>), OrbInfoError>
+    {
+        let connection = Connection::session().await.map_err(OrbInfoError::ZbusErr)?;
+
+        let auth_token_proxy = AuthTokenManagerProxy::new(&connection)
+            .await
+            .map_err(OrbInfoError::ZbusErr)?;
+
+        let initial_value = auth_token_proxy.token().await.unwrap();
+        let (send, recv) = watch::channel(initial_value);
+        let auth_token_proxy = auth_token_proxy.clone();
+        let join_handle =
+            tokio::task::spawn(Self::task_inner(auth_token_proxy, send, cancel_token));
+        Ok((join_handle, recv))
+    }
+
+    async fn task_inner(
+        auth_token_proxy: AuthTokenManagerProxy<'_>,
+        send: watch::Sender<String>,
+        cancel_token: tokio_util::sync::CancellationToken,
+    ) {
```

Comment:
Should return `color_eyre::Result<()>` so that we can allow the errors to get propagated via the `JoinHandle<Result<()>>`